lib/puppet/type/service.rb

```diff
@@ -78,14 +78,8 @@ def retrieve
         provider.delayed_start
       end
 
-      # This only makes sense on systemd systems. Static services cannot be enabled
-      # or disabled manually.
-      def insync?(current)
-        if provider.respond_to?(:cached_enabled?) && provider.cached_enabled? == 'static'
-          Puppet.debug("Unable to enable or disable static service #{@resource[:name]}")
-          return true
-        end
-
+      def insync?(current)                                                     
+        return provider.enabled_insync?(current) if provider.respond_to?(:enabled_insync?)
```

Comment:
Before this change `super(current)` was always called, now is not called of provider responds to `enabled_insync?`. Is this intended?